R/length.R

```diff
@@ -1,18 +1,49 @@
 #' Assign SAS Length
 #'
-#' Assigns SAS length from a variable level metadata to a given data frame.
+#' Assigns SAS length from a variable level metadata to a given data frame. If a
+#' length isn't present for a variable the length value is set to 200 for
+#' character columns, and 8 for non-character columns. This value is stored in
+#' the 'width' attribute of the column.
 #'
 #' @param .df A data frame of CDISC standard.
-#' @param metadata A data frame containing variable level metadata.
-#' @param domain A character value to subset the `.df`. If `NULL`(default), uses
-#'   `.df` value as a subset condition.
-#' @param verbose The action the function takes when a length isn't found in
-#'   metadata. Options are 'stop', 'warn', 'message', and 'none'
-#' @param metacore `r lifecycle::badge("deprecated")` Previously used to pass metadata now renamed with `metadata`
+#' @param metadata A data frame containing dataset/variable level metadata. See
+#'   'Metadata' section for details.
+#' @param domain Appropriate CDSIC dataset name, e.g. ADAE, DM. Used to subset
+#'   the metadata object. If none is passed, then name of the dataset passed as
+#'   .df will be used.
+#' @param verbose The action this function takes when an action is taken on the
+#'   dataset or function validation finds an issue. See 'Messaging' section for
+#'   details. Options are 'stop', 'warn', 'message', and 'none'
+#' @param metacore `r lifecycle::badge("deprecated")` Previously used to pass
+#'   metadata now renamed with `metadata`
+#'
+#' @section Messaging: `length_log` is the primary messaging tool for
+#'   `xportr_length`. If there are any columns present in the '.df' that are not
+#'   noted in the metadata, they cannot be assigned a length and a message will
+#'   be generated noting the number or variables that have not been assigned a
+#'   length.
+#'
+#'   If variables were not found in the metadata and the value passed to the
+#'   'verbose' argument is 'stop', 'warn', or 'message', a message will be
+#'   generated detailing the variables that were missing in metadata.
+#'
+#' @section Metadata: The argument passed in the 'metadata' argument can either
+#'   be a metacore object, or a data.frame containing the data listed below. If
```

Comment:
```suggestion
#'   be a `{metacore}` object, or a data.frame containing the data listed below. If
```
Then it links to the metacore pkgdown site